tests/src/analysis/testqgsprocessingalgs.cpp

```diff
@@ -235,6 +236,32 @@ QVariantMap pkgAlg( const QStringList &layers, const QString &outputGpkg, bool o
   return package->run( parameters, *context, &feedback, ok );
 }
 
+void TestQgsProcessingAlgs::saveFeaturesAlg()
+{
+  QString outputGeoJson = QDir::tempPath() + "/savefeatures_alg.geojson";
+
+  if ( QFile::exists( outputGeoJson ) )
+    QFile::remove( outputGeoJson );
+
+  QString dataDir( TEST_DATA_DIR ); //defined in CmakeLists.txt
+
+  QVariantMap parameters;
+  parameters.insert( QStringLiteral( "INPUT" ), dataDir + "/polys.shp" );
+  parameters.insert( QStringLiteral( "LAYER_OPTIONS" ), QStringLiteral( "COORDINATE_PRECISION=5" ) );
+  parameters.insert( QStringLiteral( "OUTPUT" ), outputGeoJson );
+
+
+  const QgsProcessingAlgorithm *saveFeatures( QgsApplication::processingRegistry()->algorithmById( QStringLiteral( "native:savefeatures" ) ) );
+  std::unique_ptr< QgsProcessingContext > context = qgis::make_unique< QgsProcessingContext >();
+  context->setProject( QgsProject::instance() );
+
+  QgsProcessingFeedback feedback;
+  bool ok = false;
+  QVariantMap outputs = saveFeatures->run( parameters, *context, &feedback, &ok );
+  QCOMPARE( ok, true );
+  QCOMPARE( outputs.value( QStringLiteral( "OUTPUT" ) ).toString(), outputGeoJson );
+}
```

Comment:
Better also read back in the created file and confirm that the precision argument has been correctly applied